### DISCUSSION TITLE
Is there a way to show diagnostics for test/IntegrationTests without need to enter the test folder?

### DISCUSSION THREAD

**Discussion by bartoszb-rgb** [scalameta/nvim-metals#594] [Q&A]
Hey,
Every time did some changes to classes in `src` I would expect all usages it tests also to display as compilation errors.
To actually display those I need to enter any Scala file in the `test` folder to compile it to display those, which is kid of annoying.

Is there a way to show all related diagnostics?

**Comment by ckipp01** [comment]
Hey @bartoszb-rgb, good question. So by default Metals doesn't compile every single build target in your project, only the ones you're working on. This will save a lot of compilation cycles providing a better experience for the majority of the time. If you do want to compile the targets that are relying on the target that you're currently work on, you can use the cascade compile which is available via `:MetalsCompileCascade` or via the lua api as well.
